libraries/amqp/include/eosio/amqp/amqp_handler.hpp

```diff
@@ -106,21 +89,81 @@ class amqp {
             } );
    }
 
+   /// Explicitly stop thread pool execution
+   /// Thread safe
```

Comment:
Concurrent calls are not safe.
Thread 1: posts to strand and blocks on future
Thread 2: posts to strand and blocks on future
Thread 1: stops thread pool
Thread 2: never wakes up